source/MRViewer/MRViewer.cpp

```diff
@@ -760,33 +760,83 @@ bool Viewer::isSupportedFormat( const std::filesystem::path& mesh_file_name )
     return false;
 }
 
-bool Viewer::loadFile( const std::filesystem::path & path )
+bool Viewer::loadFiles( const std::vector< std::filesystem::path>& filesList )
 {
-    std::string ext = utf8string( path.extension() );
-    for ( auto& c : ext )
-        c = ( char )tolower( c );
-    bool sceneFile = false;
-    for ( auto& filter : SceneFileFilters )
-    {
-        if ( filter.extension.find( ext ) != std::string::npos )
-            sceneFile = true;
-    }
-
-    auto res = loadObjectFromFile( path );
-    if ( !res.has_value() )
+    if ( filesList.empty() )
         return false;
-    if ( sceneFile )
-    {
-        auto newRoot = (*res)[0];
-        std::swap( newRoot, SceneRoot::getSharedPtr() );
-        getViewerInstance().onSceneSaved( path );
-    }
-    else
+
+    ProgressBar::orderWithMainThreadPostProcessing( "Open files", [filesList] ()->std::function<void()>
```

Comment:
mb check that graphics is available for progress bar? (it may be not)

Reply:
fixed